src/main/java/com/gravity9/mongocdc/MongoChangeStreamWorker.java

```diff
@@ -82,7 +87,11 @@ public void run() {
 
 		ChangeStreamIterable<Document> watch = collection.watch(List.of(
 				Aggregates.match(
-					expr(eq(mod(divide(toLong(toDate())), partitionNumbers), partition))
+					or(List.of(
+							partitionMatchExpression(toDateFullDocumentId(), partitionNumbers, partition),
```

Comment:
question: are we sure we won't get overlap between partitions if we use OR here? I assume we only get one or the other (full document ID or document key) and that is why it's okay?

Reply:
If both fields are available, they will be equal so event will be received by a single partition handler, so there should be no problem.

Reply:
Based on the docs (https://www.mongodb.com/docs/v5.2/reference/change-events/) `documentKey` is `A document that contains the _id of the document created or modified by the insert, replace, delete, update operations (i.e. CRUD operations).`
It looks fine for me.